packages/lumx-react/src/components/grid-column/GridColumn.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { forwardRef, ReactElement, ReactNode } from 'react';
+
+import isInteger from 'lodash/isInteger';
+
+import classNames from 'classnames';
+
+import { Comp, GenericProps } from '@lumx/react/utils/type';
+import { getRootClassName, handleBasicClasses } from '@lumx/react/utils/className';
+import { Size } from '@lumx/react';
+
+export type GridColumnGapSize = Extract<Size, 'tiny' | 'regular' | 'big' | 'huge'>;
+
+/**
+ * Defines the props of the component.
+ */
+export interface GridColumnProps extends GenericProps {
+    /** Customize the root element. */
+    as?: React.ElementType;
+    /** Children elements. */
+    children?: ReactNode;
+    /** Space between columns and rows. */
+    gap?: GridColumnGapSize;
+    /** Ideal number of columns. */
+    maxColumns?: number;
+    /** Minimum width for each item, reduce the number of column if there is not enough space. */
+    itemMinWidth?: number;
+}
+
+/**
+ * Component display name.
+ */
+const COMPONENT_NAME = 'GridColumn';
+
+/**
+ * Component default class name and class prefix.
+ */
+const CLASSNAME = getRootClassName(COMPONENT_NAME);
+
+/**
+ * Component default props.
+ */
+const DEFAULT_PROPS: Partial<GridColumnProps> = {};
+
+/**
+ * The GridColumn is a layout component that can display children in a grid
+ * with custom display properties. It also comes with a responsive design,
+ * with a number of column that reduce when there is not enough space for each item.
+ *
+ * @param props Component props.
+ * @param ref Component ref.
+ * @return React element.
+ */
+export const GridColumn: Comp<GridColumnProps> = forwardRef(
+    (props, ref): ReactElement => {
+        const {
+            as: Component = 'div',
+            gap,
+            maxColumns,
+            itemMinWidth,
+            children,
+            className,
+            style = {},
+            ...forwardedProps
+        } = props;
+
+        return (
+            <Component
+                {...forwardedProps}
+                ref={ref as React.Ref<any>}
+                className={classNames(
+                    className,
+                    handleBasicClasses({
+                        prefix: CLASSNAME,
+                    }),
+                )}
+                style={{
+                    ...style,
+                    ['--lumx-grid-column-item-min-width' as any]: isInteger(itemMinWidth) && `${itemMinWidth}px`,
+                    ['--lumx-grid-column-columns' as any]: maxColumns,
+                    ['--lumx-grid-column-gap' as any]: gap && `var(--lumx-spacing-unit-${gap})`,
```

Comment:
I rather have the gap as a classique modifier class like in flexbox https://github.com/lumapps/design-system/blob/05042e921207cafffc4e0b5a82b741f76efd06b3/packages/lumx-react/src/components/flex-box/FlexBox.tsx#L85
Instead of a CSS variable since it's a finite list of values

Reply:
I'll still need the CSS variable, for the `grid-template-column` calculation, since we take into account the gap between items.